NFC_EEPROM/source/target/TARGET_ST25DV/eeprom_driver/st25dv_driver.h

```diff
@@ -0,0 +1,351 @@
+} TT5_MagicNumber_t;
+
+/**
+  * @brief  Type5 Tag Capability Container structure.
+  */
+typedef struct
+{
+  TT5_MagicNumber_t MagicNumber;  /**< CCfile[0]: Magic Number should be E1h or E2h (for extended API) */
+  uint8_t Version;                /**< CCfile[1]: Capability container version (b7-b4) and access conditions (b3-b0) */
+  uint8_t MemorySize;             /**< CCfile[2]: Memory size, expressed in 8 bytes blocks, set to 0 if tag size is greater than 16kbits. */
+  uint8_t TT5Tag;                 /**< CCfile[3]: Additionnal information on the Type5 Tag:\n
+                                                  b0: supports `read multiple block` commands\n
+                                                  b1: RFU\n
+                                                  b2: RFU\n
+                                                  b3: supports `lock block` commands\n
+                                                  b4: requires the `special frame` format
+                                    */
+  uint8_t rsved1;                 /**< RFU */
+  uint8_t rsved2;                 /**< RFU */
+  uint16_t ExtMemorySize;         /**< CCfile[6],CCfile[7]: Memory size, expressed in 8 bytes blocks, when tag size is greater than 16kbits. */
+  TT5_State State;                /**< Indicates if a NDEF message is present. */
+  uint32_t NDEF_offset;           /**< Indicates the address of a NDEF message in the tag. */
+}sCCFileInfo;
+
+/** @brief Type5 Tag Type-Length-Value structure as defined by the NFC Forum */
+typedef struct
+{
+  uint8_t   Type;     /**< NFC Forum message Type */
+  uint8_t   Length;   /**< Message length if lesser than 255 bytes */
+  uint16_t  Length16; /**< Message length if greater than or equal to 255 bytes */
+} TT5_TLV_t;
+
+class ST25dvDriver : public NFCEEPROMDriver {
+
+public:
+    /** Create the driver, default pin names will be used appropriate for the board.
+     *  @param i2c_data_pin I2C data pin name.
+     *  @param i2c_clock_pin I2C clock pin name.
+     *  @param gpo_pin I2C GPO pin name.
+     *  @param rf_disable_pin pin name for breaking the RF connection.
+     */
+    ST25dvDriver(PinName i2c_data_pin = ST25DV_I2C_SDA_PIN,
+                PinName i2c_clock_pin = ST25DV_I2C_SCL_PIN,
+                PinName led1_pin = ST25DV_LED1_PIN,
+                PinName led2_pin = ST25DV_LED2_PIN,
+                PinName led3_pin = ST25DV_LED3_PIN,
+                PinName lpd_pin = ST25DV_LPD_PIN,
+                PinName gpo_pin = ST25DV_GPO_PIN);
+
+    virtual ~ST25dvDriver() { }
+
+    /** @see NFCEEPROMDriver::reset
+     */
+    virtual void reset() {
+        printf("reset\r\n");
+        begin();
+    }
+
+    /** @see NFCEEPROMDriver::get_max_size
+     */
+    virtual size_t read_max_size() {
+        return MAX_NDEF_SIZE;
+    }
+
+    /** @see NFCEEPROMDriver::start_session
+     */
+    virtual void start_session(bool force = true) {
+        printf("start_session\r\n");
+        if(_is_device_inited) {
```

Comment:
Presumably the NFC EEPROM needs to disable the RF interface (`RF_DISABLE` set to 1 in `RF_MNGT_Dyn`), unless a RF transaction is already ongoing and force is false